extensions/dicom-microscopy/src/DicomMicroscopyViewport.tsx

```diff
@@ -1,318 +1,217 @@
-              },
-            ],
-          };
-        }
-        const image = new metadataUtils.VLWholeSlideMicroscopyImage({
-          metadata: inst,
         });
 
-        const imageFlavor = image.ImageType[2];
-        if (imageFlavor === 'VOLUME' || imageFlavor === 'THUMBNAIL') {
-          volumeImages.push(image);
+        // format metadata for microscopy-viewer
+        const options = {
+          client,
+          metadata: volumeImages,
+          retrieveRendered: false,
+          controls: ['overview', 'position'],
+        };
+
+        const viewer = new microscopyViewer(options);
+
+        if (overlayElement && overlayElement.current && viewer.addViewportOverlay) {
+          viewer.addViewportOverlay({
+            element: overlayElement.current,
+            coordinates: [0, 0], // TODO: dicom-microscopy-viewer documentation says this can be false to be automatically, but it is not.
+            navigate: true,
+            className: 'OpenLayersOverlay',
+          });
         }
-      });
 
-      // format metadata for microscopy-viewer
-      const options = {
-        client,
-        metadata: volumeImages,
-        retrieveRendered: false,
-        controls: ['overview', 'position'],
-      };
+        viewer.render({ container });
 
-      this.viewer = new microscopyViewer(options);
+        const { StudyInstanceUID, SeriesInstanceUID } = displaySet;
 
-      if (this.overlayElement && this.overlayElement.current && this.viewer.addViewportOverlay) {
-        this.viewer.addViewportOverlay({
-          element: this.overlayElement.current,
-          coordinates: [0, 0], // TODO: dicom-microscopy-viewer documentation says this can be false to be automatically, but it is not.
-          navigate: true,
-          className: 'OpenLayersOverlay',
-        });
-      }
-
-      this.viewer.render({ container });
-
-      const { StudyInstanceUID, SeriesInstanceUID } = displaySet;
+        const managedViewer = microscopyService.addViewer(
+          viewer,
+          viewportId,
+          container,
+          StudyInstanceUID,
+          SeriesInstanceUID
+        );
 
-      this.managedViewer = this.microscopyService.addViewer(
-        this.viewer,
-        this.props.viewportId,
-        container,
-        StudyInstanceUID,
-        SeriesInstanceUID
-      );
+        managedViewer.addContextMenuCallback((event: Event) => {
+          // TODO: refactor this after Bill's changes on ContextMenu feature get merged
+          // const roiAnnotationNearBy = this.getNearbyROI(event);
+        });
 
-      this.managedViewer.addContextMenuCallback((event: Event) => {
-        // TODO: refactor this after Bill's changes on ContextMenu feature get merged
-        // const roiAnnotationNearBy = this.getNearbyROI(event);
-      });
-    };
+        setViewer(viewer);
+        setManagedViewer(managedViewer);
+      };
 
-    this.microscopyService.clearAnnotations();
+      microscopyService.clearAnnotations();
```

Comment:
You cannot rely on the global microscopy service due to the how the libraries get loaded.  Please revert these changes.

Reply:
Never mind, I see what you are doing here.